engine/swarm/networking.md

```diff
@@ -125,179 +126,228 @@ $ docker network inspect my-network
+
+```bash
+$ docker service create \
+  --replicas 3 \
+  --name my-web \
+  --network my-network \
+  nginx
+```
 
-2.  Lookup the node where `my-busybox` is running:
+Service task containers connected to an overlay network can communicate with
+each other across it.
 
-    ```bash
-    $ docker service ps my-busybox
+To see which networks a service is connected to, use `docker service ls` to find
+the name of the service, then `docker service ps <service-name>` to list the
+networks. Alternately, to see which task containers are connected to a network,
+use `docker network inspect <network-name>`. You can run these commands from
+any swarm node which is joined to the swarm and is in a `running` state.
 
-    NAME                                    IMAGE    NODE   DESIRED STATE  CURRENT STATE          ERROR
-    my-busybox.1.1dok2cmx2mln5hbqve8ilnair  busybox  node1  Running        Running 5 seconds ago
-    ```
+### Configure service discovery
 
-3.  From the node where the busybox task is running, open an interactive shell to
-the busybox container:
+**Service discovery** is the mechanism Docker uses to route a request from your
+service's external clients to an individual swarm node, without the client
+needing to know how many nodes are participating in the service or their
+IP addresses or ports. You don't need to publish ports which are used between
+services on the same network. For instance, if you have a WordPress service
+which stores its data in a MySQL service, and they are connected to the same
+overlay network, you do not need to publish the MySQL port to the client, only
+the WordPress HTTP port.
 
-    ```bash
-    $ docker exec -it my-busybox.1.1dok2cmx2mln5hbqve8ilnair /bin/sh
-    ```
+Service discovery can work in two different ways, using a virtual IP (VIP) or
+DNS round robin (DNSRR). You can configure this per service.
 
-    You can deduce the container name as `<TASK-NAME>`+`<ID>`. Alternatively,
-    you can run `docker ps` on the node where the task is running.
+- By default, when you attach a service to a network, Docker assigns the service
+  a virtual IP (VIP), which is the "front end" for clients to reach the service.
+  Docker keeps a list of all worker nodes in the service, and routes requests
+  between the client and one of the nodes. Each request from the client might be
+  routed to a different node. This mechanism uses a protocol called `gossip`.
 
-4.  From inside the busybox container, query the DNS to view the VIP for the
-`my-web` service:
+- If you configure a service to use DNS round-robin (DNSRR) service discovery,
+  there is not a single virtual IP. Instead, Docker sets up DNS entries for the
+  service such that a DNS query for the service name returns a list of IP
+  addresses, and the client connects directly to one of these.
 
-    ```bash
-    $ nslookup my-web
+  You should only use DNSRR service discovery if your service uses an external
```

Comment:
I think the original sentence is more accurate. "DNS round-robin is useful in cases where you want to use your own load balancer."